app/src/main/java/com/sopt/now/presentation/auth/login/LoginViewModel.kt

```diff
@@ -0,0 +1,30 @@
+package com.sopt.now.presentation.auth.login
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.sopt.now.domain.model.LoginInfo
+import com.sopt.now.domain.usecase.LoginUseCase
+import com.sopt.now.presentation.auth.AuthStatus
+import com.sopt.now.presentation.auth.AuthState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginViewModel @Inject constructor(
+    private val loginUseCase: LoginUseCase
+) : ViewModel() {
+    private val _loginState = MutableStateFlow(AuthState(AuthStatus.UNINITIALIZED,"",false))
+    val loginState: StateFlow<AuthState> = _loginState.asStateFlow()
+
+    fun login(authenticationId: String, password: String) {
+        _loginState.value = AuthState(AuthStatus.PROCESSING, "", true)
+        viewModelScope.launch {
+            val result = loginUseCase.execute(LoginInfo(authenticationId, password))
```

Comment:
여기까지 그냥 내려오다가 지금 생각났는데, Kotlin의 `invoke`에 대해서도 한 번 공부해보시면 좋을 것 같습니다. UseCase에서 invoke를 사용한다면, 코드가 아래와 같이 바뀌게 될 것입니다. 솔직히 근데 안드로이드를 안 한 지 좀 돼서 가물가물하네요. 틀렸다면 지적 부탁드립니다.
```suggestion
            val result = loginUseCase(LoginInfo(authenticationId, password))
```